src/pushsource/_impl/backend/koji_source.py

```diff
@@ -509,6 +521,127 @@ def _push_items_from_container_build(self, nvr, meta):
+
+        # Supported types of VMI archives
+        ami_types = ["raw", "raw-xz"]
+        azure_types = ["vhd", "vhd-compressed"]
+        cloud_types = {}
+
+        for k in ami_types:
+            cloud_types.setdefault(k, AmiPushItem)
+
+        for k in azure_types:
+            cloud_types.setdefault(k, VHDPushItem)
+
+        # Collect all image archives which are not containers (tar)
+        # FIXME: Need to verify whether it's better to limit only to
+        # known extensions (like VHD, AMI) or not
+        vmi_archives = [
+            elem
+            for elem in archives
+            if elem.get("btype") == "image" and elem.get("type_name") != "tar"
+        ]
+
+        out = []
+
+        # Generate the respective PushItem type for each archive
+        for archive in vmi_archives:
+            path = self._pathinfo.typedir(meta, archive["btype"])
+            item_src = os.path.join(path, archive["filename"])
+
+            extra = archive.get("extra") or {}
+            image = extra.get("image") or {}
+            arch = image.get("arch")
+
+            klass = cloud_types.get(archive["type_name"]) or VMIPushItem
+
+            rel_klass = AmiRelease if klass == AmiPushItem else VMIRelease
+
+            # FIXME
+            # Missing information in "release":
+            # - product (it's using a workaround on 'name' based on internal naming convention,
+            #            it is, the last substring from dash is the cloud name, eg.: rhel-azure)
+            # - respin  (it's using the 'release' as workaround)
+            # - base_product
+            # - base_version
+            # - variant
+            # - type
+            splitted_name = meta["name"].split("-")
+            if len(splitted_name) > 1:
+                product = "-".join(splitted_name[:-1])
+            else:
+                product = splitted_name[0]
+
+            release = rel_klass(
+                arch=arch,
+                date=meta.get("completion_time", "").split(" ")[0],
+                product=product.upper(),
+                version=meta["version"],
+                respin=int(meta["release"]),
+            )
+
+            out.append(
+                # FIXME: Is it better to ask osbuild to export the missing info in metadata?
+                #
+                # Remaining data to be gattered from an external service:
+                #
+                # AmiPushItem:
+                #   - type
+                #   - region
+                #   - virtualization
+                #   - volume
+                #   - root_device
+                #   - sriov_net_support
+                #   - ena_support
+                #   - billing_codes
+                #
+                # VHDPushItem:
+                #   - generation
+                #   - sku_id
+                #   - support_legacy
+                #   - legacy_sku_id
+                #   - disk_version
+                #   - recommended_sizes
+                #   - sas_uri
```

Comment:
@Tojaj @lslebodn these are the missing metadata from Koji. 

For `AmiPushItem` they're available from Staged Source but here we can't find them.

For `VHDPushItem` we also need everything but `sas_uri` which will be generated after the upload.

Reply:
We have 2 options here:

1. Ask osbuild team to provide the missing metadata depending on the type of image (VHD / RAW)
2. Include the missing pieces in StArMap but it will require the declaration on  `meta` for all affected policies.

Particularly, if possible I would like to have the option 1 in place as our source of information must be Koji and not another service... WDYT?